src/wsdiscovery.ts

```diff
@@ -27,92 +36,176 @@ export default class WsDiscovery extends EventEmitter {
             `WsDiscovery initilized with the following options: ${JSON.stringify(this.opts)}`,
             WsDiscovery.name
         );
-        this.bindSocket();
+        this.initNetworkInterfaces();
+        this.watchNetworkInterfaces();
     }
 
-    bindSocket(): void {
-        this.socket = this.opts.socket || dgram.createSocket('udp4');
-        this.socket.bind(() => {
-            const map = this.findL1HostInterface();
-            if ((map.ip && map.interface) || this.opts.targetInterfaceAddr) {
-                const targetInterfaceAddr = this.opts.targetInterfaceAddr || map.ip;
-                this.socket.setMulticastInterface(targetInterfaceAddr);
-                Logger.debug(
-                    `Set default outgoing multicast interface of the socket to interface with address ${targetInterfaceAddr}`,
-                    WsDiscovery.name
-                );
-            }
-            if (!this.interfaceWatcher) {
+    initSocketConnection(
+        interfaceMap: InterfaceMap = { interfaceName: 'default', ip: undefined }
+    ): dgram.Socket {
+        Logger.debug(
+            `Discovery probe messages bound to target interface [${interfaceMap.interfaceName}] on addr ${interfaceMap.ip}`,
+            WsDiscovery.name
+        );
+        const socketConnection = dgram.createSocket('udp4');
+        this.bindSocket(interfaceMap, socketConnection);
+        return socketConnection;
+    }
+
+    initNetworkInterfaces(): void {
+        const probeEntireNetwork =
+            this.opts.probeEntireNetwork || process.env.HUDDLY_WSDD_PROBE_ENTIRE_NETWORK;
+        const networkInterfaces = this.findNetworkInterfaces();
+        if (this.opts.targetInterfaceAddr || this.opts.targetInterfaceName) {
+            const { interfaceName, ip } = networkInterfaces.targetInterface;
+            if (!ip) {
                 Logger.debug(
-                    `Initiating interface watch for ${map.interface || 'default'}`,
+                    `Unable to locate target interface with ip address ${this.opts.targetInterfaceAddr} or name ${this.opts.targetInterfaceName}`,
                     WsDiscovery.name
                 );
-                this.watchInterface(map.interface || 'default');
+                return;
             }
-        });
-        this.socket.on('error', err => {
-            this.emit('ERROR', err);
-        });
-    }
+            this.socketConnections[
+                networkInterfaces.targetInterface.interfaceName
```

Comment:
Either disregard the previous comment or reference here the destructed `interfaceName` instead of doing `networkInterfaces.targetInterface.interfaceName`